src/settingsmanager.cpp

```diff
@@ -81,10 +81,26 @@ bool SettingsManager::setDownloadDir(QString downloadDir)
     return true;
 }
 
+bool SettingsManager::setMonitorDir(QString monitorDir)
+{
+    m_monitorDir = monitorDir;
+    m_settings.setValue("monitor/dir", monitorDir);
+    emit(monitorDirChanged(monitorDir));
+    return true;
+}
+
+void SettingsManager::setMonitorDirZims(QStringList zimList)
+{
+    m_monitorDirZims = zimList;
+    m_settings.setValue("monitor/zims", QVariant::fromValue(zimList));
+}
+
 void SettingsManager::initSettings()
 {
     m_kiwixServerPort = m_settings.value("localKiwixServer/port", 8080).toInt();
     m_zoomFactor = m_settings.value("view/zoomFactor", 1).toDouble();
     m_downloadDir = m_settings.value("download/dir", QString::fromStdString(kiwix::getDataDirectory())).toString();
     m_kiwixServerIpAddress = m_settings.value("localKiwixServer/ipAddress", QString("0.0.0.0")).toString();
+    m_monitorDir = m_settings.value("monitor/dir", QString("")).toString();
+    m_monitorDirZims = m_settings.value("monitor/zims", QStringList()).toStringList();
```

Comment:
Why do you need to store the list of ZIM files in settings?

Reply:
need to check on startup also, part of this PR is also to fix https://github.com/kiwix/kiwix-desktop/issues/644

Reply:
OK. I was under the wrong impression that `m_settings.value(X, V)` sets the setting `X` to `V` (I now understand that `V` is the default value for the case when `X` is not yet defined). Then persisting the list of ZIM files from the monitored directory makes sense, though with the following reservation:

Ideally (and in accordance with the [DRY principle](https://en.wikipedia.org/wiki/Don't_repeat_yourself)) on startup we should build the list of known files from the monitored directory by filtering the library XML file. The current solution can be an acceptable hack if the right approach is significantly more difficult to implement. Then, at least, the reason should be noted in a comment.

Reply:
Thank you, it makes it so much clear